File: Code/RenameStrategyIMGPNG.cs
using System.IO;
using System.Text.RegularExpressions;

namespace Code
{

    public class RenameStrategyIMGPNG : RenameStrategyBase
    {

        private const string mask = @"img_\d{8}_.*\.png";

        public override bool CanRename(string fullFileName)
        {
            var fileName = Path.GetFileName(fullFileName);
            if (!Regex.IsMatch(fileName, mask, RegexOptions.IgnoreCase)) return false;

            return true;
        }

        protected override string generateNewFileName(string current) => current.Substring(4);
    }
}

File: src/main/java/net/siudek/RenameStrategyImgPng.java
package net.siudek;

import org.springframework.stereotype.Component;

@Component
class RenameStrategyImgPng extends RenameStrategyBaseOnRegex {

  private final static String FILENAME_REGEX = "img_\\d{8}_.*\\.png";

  public RenameStrategyImgPng() {
    super(FILENAME_REGEX);
  }

  @Override
  protected String generateNewFileName(String current) {
    return current.substring(4);
  }
}
